examples/flutter_gallery/lib/demo/material/chip_demo.dart

```diff
@@ -4,106 +4,93 @@
-  'customer': 'shrine/vendors/16c477b.jpg',
-};
-
-final Map<String, Set<String>> _toolActions = <String, Set<String>>{
-  'hammer': new Set<String>()..addAll(<String>['flake', 'fragment', 'splinter']),
-  'chisel': new Set<String>()..addAll(<String>['flake', 'nick', 'splinter']),
-  'fryer': new Set<String>()..addAll(<String>['fry']),
-  'fabricator': new Set<String>()..addAll(<String>['solder']),
-  'customer': new Set<String>()..addAll(<String>['cash in', 'eat']),
-};
-
-final Map<String, Set<String>> _materialActions = <String, Set<String>>{
-  'poker': new Set<String>()..addAll(<String>['cash in']),
-  'tortilla': new Set<String>()..addAll(<String>['fry', 'eat']),
-  'fish and': new Set<String>()..addAll(<String>['fry', 'eat']),
-  'micro': new Set<String>()..addAll(<String>['solder', 'fragment']),
-  'wood': new Set<String>()..addAll(<String>['flake', 'cut', 'splinter', 'nick']),
-};
-
-class _ChipsTile extends StatelessWidget {
-  const _ChipsTile({
-    Key key,
-    this.label,
-    this.children,
-  }) : super(key: key);
+  @override
+  _ChipColumnState createState() => new _ChipColumnState();
+}
 
-  final String label;
-  final List<Widget> children;
+class _ChipColumnState extends State<ChipColumn> {
+  bool _showChip = true;
+  bool _showInputChip = true;
+  bool _choiceSelected = false;
+  bool _filterSelected = false;
 
-  // Wraps a list of chips into a ListTile for display as a section in the demo.
   @override
   Widget build(BuildContext context) {
-    return new ListTile(
-      title: new Padding(
-        padding: const EdgeInsets.only(top: 16.0, bottom: 4.0),
-        child: new Text(label, textAlign: TextAlign.start),
+    final List<Widget> chips = <Widget>[
+      ChoiceChip(
+        label: const Text('ChoiceChip'),
+        selected: _choiceSelected,
+        onSelected: (bool value) {
+          setState(() {
+            _choiceSelected = value;
+          });
+        },
+      ),
+      new FilterChip(
+        label: const Text('FilterChip'),
+        selected: _filterSelected,
+        onSelected: (bool value) {
+          setState(() {
+            _filterSelected = value;
+          });
+        },
       ),
-      subtitle: children.isEmpty
-          ? new Center(
-              child: new Padding(
-                padding: const EdgeInsets.all(8.0),
-                child: new Text(
-                  'None',
-                  style: Theme.of(context).textTheme.caption.copyWith(fontStyle: FontStyle.italic),
-                ),
-              ),
-            )
-          : new Wrap(
-              children: children
-                  .map((Widget chip) => new Padding(
-                        padding: const EdgeInsets.all(4.0),
-                        child: chip,
-                      ))
-                  .toList(),
+      new ActionChip(
+        label: const Text('ActionChip'),
+        onPressed: () {
+          setState(() {
+            Scaffold.of(context).showSnackBar(const SnackBar(
```

Comment:
Is the setState() needed here? I wouldn't think so.

Also indentation. You could write this like:
```
Scaffold.of(context).showSnackBar(
  const SnackBar(
    content: const Text('Activate!', textAlign: TextAlign.center),
  ),
);
```

Reply:
Yeah, not needed.